File: C/Funciones/isTriangle.c
//Función mencionada en Project Euler #61

int isTriangle(int n){ //Comprueba si n es un número es triangular
    int triangle = 0, i = 1, suma = i;
    if (n > 1){
        do {
            suma = i * (i + 1) / 2;
            if (suma == n) triangle = 1;
            else i++;
        } while (triangle == 0 && suma < n);
    } else triangle = 1;
    return triangle;
}

File: Python/Funciones/is_triangle.py
#Función usada en Project Euler #955

def is_triangle(n: int) -> bool: # Comprueba si n es triangular
    d = 1 + 8*n
    s = math.isqrt(d)
    return s*s == d and (s - 1) % 2 == 0
